src/services/TimerService.ts

```diff
@@ -47,27 +47,40 @@ export type TimerEventStr = "start" | "second" | "minute" | "hour" | "stop" | "a
 export class TimerService extends Service {
 
     /**
-     * An instance field containing this service's {@link NodeJS.Timeout}'s ID.
+     * Getter for {@link NodeJS.Timeout} ID.
      *
      * Can be undefined if no {@link NodeJS.Timeout} is scheduled for this {@TimerService} instance.
      */
-    private timeout?: number = undefined;
+    public get timeoutId(): number {
+        return this.timeout;
+    }
 
     /**
-     * The {@link Date} object representing the last time a timer tick happened.
+     * Convert the {@link Alarm} {@link Time}s to a {@link Map}.
      *
-     * Updated on each tick.
+     * @param alarm The {@link Alarm} whose {@link Time}s we are dealing with.
      */
-    private lastTick?: Date = undefined;
+    private static alarmTimesAsMap(alarm: Alarm): Map<AlarmEventType, Time> {
+        return new Map<AlarmEventType, Time>([
+            [ AlarmEventType.SLEEP, alarm.sleepTime ],
+            [ AlarmEventType.WAKE, alarm.wakeTime ],
+            [ AlarmEventType.GET_UP, alarm.getUpTime ]
+        ]);
+    }
 
     /**
-     * Getter for {@link NodeJS.Timeout} ID.
+     * An instance field containing this service's {@link NodeJS.Timeout}'s ID.
      *
      * Can be undefined if no {@link NodeJS.Timeout} is scheduled for this {@TimerService} instance.
      */
-    public get timeoutId(): number {
-        return this.timeout;
-    }
+    private timeout?: number = undefined;
+
+    /**
+     * The {@link Date} object representing the last time a timer tick happened.
+     *
+     * Updated on each tick.
+     */
+    private lastTick?: Date = undefined;
 
     /**
      * A map to register {@link TimerEvent}s and their respective lists of {@link TimerHandler}s.
```

Comment:
I can't complain about existing lines, but on line 112, I think we should instead have a Watcher that listens for changes to alarms in active schedules instead. Polling makes alarm activation unreliable if you're testing alarms.

Reply:
Nevermind that comment, I re-read the code and misunderstood how it worked.